plugins/inventory/upcloud.py

```diff
@@ -173,7 +211,28 @@ def _filter_servers(self):
 
             self.servers = tmp
 
+        if self.get_option("labels"):
+            display.vv("Choosing servers by labels")
+            tmp = []
+            for server in self.servers:
+                disqualified = False
+                for wanted_label in self.get_option("labels"):
+                    server_labels = _parse_server_labels(server.labels['label'])
+                    if len(server_labels) == 0:
+                        disqualified = True
+
+                    for server_label in server_labels:
+                        display.vvvv(f"Comparing {wanted_label} against {server_label}")
+                        if wanted_label not in server_label:
+                            disqualified = True
```

Comment:
If server has e.g. labels: `['foo=foo', 'bar']` and we wanted label `['bar']`, we will disqualify the server as the first hit to check `'bar' in 'foo=foo'` will fail and set disqualified to True? 🤔

Reply:
Current labels implementation checks for the existance of all labels, not if a server has some of them. Do you think it makes sense or should we change it to similar as zones where you specify multiple ones and they're all included?

Reply:
Not sure about the ansible-inventory mind model, but – to me – servers not showing up when i've put in at least 1 of the labels would be confusing 😄

Reply:
Now changed to that. It's different from how tags work there, but as we're moving away from them anyway, it's not that big of an issue.